mono/sgen/sgen-minor-scan-object.h

```diff
@@ -9,6 +9,10 @@
  * Licensed under the MIT license. See LICENSE file in the project root for full license information.
  */
 
+#ifdef __cplusplus
```

Comment:
Use `G_BEGIN_DECLS` in all of `sgen/`

Reply:
I tried and it does not always work. G_ for example isn’t for public headers. I can try again.

Reply:
If we are to be consistent, I think MONO_ is the only choice. I.e. no G_ in public headers. This will likely require adding some defines which I hadn’t done.

Reply:
extern “C” w/o macro is also a good option for consistency. Very portable no matter includes & defines.

Reply:
I appears I already use `G_BEGIN_DECLS` in all of sgen. Ok?
Similarly mini.
utils has a mix.
metadata has a mix.

Is consistency within each of the 5-ish directories (utils, metadata, mini, libgc, sgen) enough, or you require consistency across all 5?

libgc changes are needed (the pointer type  changes are negotiable, but they benefit the callers, and match mainline; the extern C is required, unless maybe libgc becomes C++ which was not my  direction).

Public headers exists.

So consistency across all 5 would be no macro or MONO_.

I can try MONO_ more consistently in utils and metadata.

I believe extern "C" a little outside of libgc was also deliberate, as the macros weren't defined in some places, and I didn't change that.

We could endevour for MONO_ to be defined more and used more; we could settle on it everywhere.

Reply:
This still needs to be fixed. I am not going to go over all the ones that needs fixing simply because you can very clearly see it in this PR. And no, not all of sgen is using `G_BEGIN_DECLS`, I assumed that would be pretty easy to see and fix.